code/script.js

```diff
@@ -229,97 +237,142 @@ const setSecret = () => {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+    } 
+    else {
+      alert(`No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  // Invoke a function to redraw the board with the remaining people
+  generateBoard();
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player first has to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+
+  // Store the interaction from the player in a variable
+  const playersGuess = confirm(`Do you want to guess ${personToConfirm}?`);
+  
+  // If the player wants to guess, invoke the checkMyGuess function
+  if (playersGuess) {
+    checkMyGuess(personToConfirm);
+  }
 }
 
-// If you confirm, this function is invoked
+// This function is invoked if player confirms they want to guess a person
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  // Check if player's guess is correct, show message in the win or lose section accordingly
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerText = `That's correct!\n\n The secret person is ${personToCheck}.`;
+  }
+  else {
+    winOrLoseText.innerText = `That's incorrect!\n\n It is not ${personToCheck}, the secret person is ${secret.name}.`;
+  }
+
+  winOrLoseSection.style.display = 'flex';
 }
 
-// Invokes the start function when website is loaded
+// Start function invoked when website is loaded
```

Comment:
I'm going to follow you here and start to rewrite the original comments, so much easier to understand for yourself!